logstash-core/lib/logstash/webservers/puma.rb

```diff
@@ -0,0 +1,67 @@
+# encoding: utf-8
+require "logstash/webserver"
+require "puma"
+require 'puma/single'
+require 'puma/binder'
+require 'puma/configuration'
+
+class LogStash::WebServer::Puma < LogStash::WebServer::Base
+
+  extend Forwardable
+
+  attr_reader :runner, :binder, :events
+
+  def_delegator :@runner, :stats, :stats
+
+  def initialize(logger, options={})
+    super(logger, options)
+    parse_options
+
+    @runner  = nil
+    @events  = Puma::Events.strings
+    @binder  = Puma::Binder.new(@events)
+    @binder.import_from_env
+  end
+
+  def run
+    setup_signals
+    log "=== puma start: #{Time.now} ==="
+
+    @runner = Puma::Single.new(self)
+    @status = :run
+    @runner.run
+    stop(:graceful => true)
+  end
+
+  def error(str)
+    logger.error(str) if logger.error?
+  end
+
+
+  def write_state
+
+  end
```

Comment:
yeah, will add a comment, is basically because of https://github.com/puma/puma/blob/master/lib/puma/single.rb#L82 providing you the option to write  the server state out somehow.